### DISCUSSION TITLE
Using StyledString to change text colors does not work well w/ themes. Is there another way?

### DISCUSSION THREAD

**Discussion by TwoClocks** [gyscos/cursive#619] [Q&A]
I have a `SelectView` and want to change various lines' color. Easy enough, wrap them in a `StyledString` with a `ColorStyle::front()` style. Except this doesn't work well with the theme/palette. This sets the foreground color and does not change it no matter what. It makes it very hard to read when they are `Highlight` or `HighlightInactive`. It's hard to pick a single color that is readable over light and dark colors.

Is there a way to get the `StyledString` to only set the color when `Primary` is asked for? Or alternative allow a `Palette` as a `Style` instead of just a front/back color?

Or am I doing this entirely the wrong way, to begin with?

**Comment by gyscos** [comment]
Right now highlighting indeed doesn't work well with themes or styled strings.

I think rather than overriding back and front color, I may change it to just swap front and back colors.

I like the idea of giving more control to this highlighting behavior, but I'm not sure assigning a full palette to each styled span is the best solution.

**Reply by TwoClocks** [reply]
yeah, the full palette to set a color seems like overkill. But the concept already exists in the code.

I guess you could look at the colors, and pick either the front or the back color depending on which would give the most contrast vs the theme chosen highlight/background at the moment..
